igvm/hypervisor.py

```diff
@@ -317,22 +323,41 @@ def vm_set_disk_size_gib(self, vm, new_size_gib):
         """Changes disk size of a VM."""
         if new_size_gib < vm.dataset_obj['disk_size_gib']:
             raise NotImplementedError('Cannot shrink the disk.')
-        with self.fabric_settings():
-            self.lv_resize(self.get_lv_by_vm(vm)['path'], new_size_gib)
-
-        self._vm_set_disk_size_gib(vm, new_size_gib)
+        volume = self.get_volume_by_vm(vm)
+        self.run('lvresize {} -L {}g'.format(volume.path(), new_size_gib))
+        self.storage_pool.refresh()
+        self._get_domain(vm).blockResize(
+            'vda',
+            new_size_gib * 1024 ** 2,  # Yes, it is in KiB
+        )
+        vm.run('xfs_growfs /')
 
-    def create_vm_storage(self, vm, lv_name, transaction=None):
+    def create_vm_storage(self, vm, transaction=None, vol_name=None):
```

Comment:
I know this is ugly. But @hasegeli is working on replacing those transactions with something more Python-native anyway, so maybe we should not care as it will go away anyway?